worlds/hades/Rules.py

```diff
@@ -0,0 +1,420 @@
+                    state._has_enough_routine_inspection(player,total_routine_inspection-2, options) and  \
+                    state._has_enough_weapons(player, options, 2))
+        set_rule(world.get_entrance("Exit Asphodel", player), lambda state: state.has("LernieVictory", player) and  \
+                    state._total_heat_level(player, min(number_items/2,20), options) and \
+                    state._has_enough_routine_inspection(player,total_routine_inspection-1, options) and  \
+                    state._has_enough_weapons(player, options, 3))
+        set_rule(world.get_entrance("Exit Elyseum", player), lambda state: state.has("BrosVictory", player) and  \
+                    state._total_heat_level(player, min(number_items*3/4,30), options) and  \
+                    state._has_enough_routine_inspection(player,total_routine_inspection, options) and  \
+                    state._has_enough_weapons(player, options, 5))
+        set_rule(world.get_location("Beat Hades", player), lambda state:  \
+                    state._total_heat_level(player, min(number_items,35), options) and  \
+                    state._has_enough_weapons(player, options, 6))
+    
+
+    forbid_important_items_on_late_styx(world, player, options)
+    world.completion_condition[player] = lambda state: state._can_get_victory(player, options)
+    
+    if (options.keepsakesanity.value==1):
+        add_rule(world.get_entrance("NPCS", player), lambda state: True)
+        add_rule(world.get_location("EurydiceKeepsake",player), lambda state:  \
+                 state._has_defeated_boss("LernieVictory", player, options))
+        add_rule(world.get_location("ThanatosKeepsake",player), lambda state:  \
+                state._has_defeated_boss("HadesVictory", player, options))
+        add_rule(world.get_location("PatroclusKeepsake",player), lambda state:  \
+                state._has_defeated_boss("BrosVictory", player, options))
+        set_keepsake_balance(world, player, location_table, options)
+    if (options.weaponsanity.value == 1):
+        add_rule(world.get_entrance("Weapon Cache", player), lambda state: True)
+    if (options.storesanity.value == 1):
+        set_store_rules(world, player, location_table, options)
+    if (options.fatesanity.value == 1):
+        set_fates_rules(world, player, location_table, options, "")
+    set_fates_rules(world, player, location_table, options, "Event")
+    
+
+    if (options.keepsakesanity.value == 1 and options.storesanity.value == 1):
+        add_rule(world.get_location("OrpheusKeepsake", player), lambda state: \
+                state.has("CourtMusicianSentenceItem",player))
+
+    visualize_regions(world.get_region("Menu", player), "my_world.puml")
+
+def set_keepsake_balance(world: MultiWorld, player: int, location_table, options):
+    set_rule(world.get_location("ZeusKeepsake", player), lambda state:   \
+            state.has("ZeusKeepsake", player))
```

Comment:
```suggestion
def set_keepsake_balance(world: "HadesWorld", player: int, location_table, options):
    set_rule(world.get_location("ZeusKeepsake"), lambda state:   \
            state.has("ZeusKeepsake", player))
```
Absolutely do not use `world: MultiWorld`, it is very confusing.